Cargo.toml

```diff
@@ -77,6 +77,10 @@ fd-lock = { version = "3", optional = true }
 # full grammar check, but also tokenization and disambiguation
 nlprule = { version = "=0.6.4", optional = true }
 
+# TODO: find way to only pull in for [[bin]], there is https://github.com/rust-lang/rfcs/pull/2887
+# but it's stale sind 2015
+syscalls = "0.5"
```

Comment:
Yeah, that's still a bit of a mess. `required-features = ["test-signal"]` is the cleanest, and only make it condiational on that to avoid pulling it in for the regular binary.